### PR TITLE
Use bumpversion to set version in application.properties

### PR DESCRIPTION
**Pull request by anguss00**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -15,6 +9,23 @@ pipeline {
...
+                    steps {
+                        sh 'bumpversion patch'
+                    }
+                }
+                stage('release') {
+                    when { branch 'master' }
+                    steps {
+                        withCredentials([usernamePassword(credentialsId: env.CREDENTIALS_ID, passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                            sh '''
+                                export VERSION=$(bump2version --list --allow-dirty release | grep new_version= | sed -r s,"^.*=",,)
```
Comment (lwgordonimos) [inline Jenkinsfile]: Can get rid of a pipe with the following:
```
bump2version --list --allow-dirty release | grep -oP '^new_version=\K.*$'
```

File: Jenkinsfile [hunk 2]
```diff
@@ -15,6 +9,23 @@ pipeline {
...
+                                git push origin master
+                                git push origin refs/tags/v$VERSION
```
Comment (lwgordonimos) [inline Jenkinsfile]: Can be written in a cleaner CLI:
`git push origin tag v$VERSION`

from git-push man page:
```tag <tag> means the same as refs/tags/<tag>:refs/tags/<tag>.```

File: Jenkinsfile [hunk 3]
```diff
@@ -15,6 +9,23 @@ pipeline {
...
+                    steps {
+                        sh 'bumpversion patch'
+                    }
+                }
+                stage('release') {
+                    when { branch 'master' }
+                    steps {
+                        withCredentials([usernamePassword(credentialsId: env.CREDENTIALS_ID, passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                            sh '''
+                                export VERSION=$(bump2version --list --allow-dirty release | grep -oP '^new_version=\K.*$')
```
Comment (lwgordonimos) [inline Jenkinsfile]: As discussed, I think we should move all the version bumping scripts into a `bumpversion.sh` script in the repo for consistency and easier testing.

File: Dockerfile [hunk 4]
```diff
@@ -12,11 +11,23 @@ ENV PATH $GRAILS_HOME/bin:$PATH
 RUN apt-get update && apt-get install -y --no-install-recommends software-properties-common \
     && add-apt-repository ppa:openjdk-r/ppa \
     && apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
```
Comment (lwgordonimos) [inline Dockerfile]: Not sure why build-essential, libudunits2-dev and python-dev are here?

Also, `git-core` is an old metapackage which just points to `git`, so just use that directly.

**Comment by anguss00** [comment]
@lwgordonimos ready for your review

**Pull request merged**